### ISSUE TITLE
The values of running variance from pretrained file are negative

### ISSUE THREAD

**Issue by jason2718** [hufu6371/DORN#25]

**Comment by angshine** [comment]
But it seems that kitti_demo.py can work properly with that negative running_var in batchnorm, why is that?

**Comment by angshine** [comment]
> But it seems that kitti_demo.py can work properly with that negative running_var in batchnorm, why is that?

While, it turns out that I referenced the wrong caffe layer in when parsing the caffe weights. The order of blobs can be found in[(bn_layer.cpp)](https://github.com/hufu6371/DORN/blob/master/caffe/src/caffe/layers/bn_layer.cpp).
``` cpp
    // slope
    this->blobs_[0].reset(new Blob<Dtype>(shape));
    shared_ptr<Filler<Dtype> > slope_filler(GetFiller<Dtype>(
        this->layer_param_.bn_param().slope_filler()));
    slope_filler->Fill(this->blobs_[0].get());
    // bias
    this->blobs_[1].reset(new Blob<Dtype>(shape));
    shared_ptr<Filler<Dtype> > bias_filler(GetFiller<Dtype>(
        this->layer_param_.bn_param().bias_filler()));
    bias_filler->Fill(this->blobs_[1].get());
    // moving average mean
    this->blobs_[2].reset(new Blob<Dtype>(shape));
    caffe_set(this->blobs_[2]->count(), Dtype(0),
        this->blobs_[2]->mutable_cpu_data());
    // moving average variance
    this->blobs_[3].reset(new Blob<Dtype>(shape));
    caffe_set(this->blobs_[3]->count(), frozen_ ? Dtype(1) : Dtype(0),
        this->blobs_[3]->mutable_cpu_data());
```
